aptos-move/move-examples/fungible_asset/stablecoin/sources/fiat_token.move

```diff
@@ -0,0 +1,420 @@
+        
+        // retrieve the refs using the `Metadata` object
+        let metadata_object = get_metadata_object();
+        let fiat_token_config = authorized_borrow_refs(minter, metadata_object);
+
+        // mint the tokens and deposit the minted tokens to the `to` address
+        let to_wallet = primary_fungible_store::ensure_primary_store_exists(to, metadata_object);
+        let fiat_token = fungible_asset::mint(&fiat_token_config.mint_ref, amount);
+        fungible_asset::deposit_with_ref(&fiat_token_config.transfer_ref, to_wallet, fiat_token);
+
+        // decrease minter allowance by amount
+        decrement_minter_allowance(signer::address_of(minter), amount);
+
+        // TODO: emit minting event (??) or is FA events sufficient, indexing could be done that way
+    }
+
+    /// Burn fiat tokens from the minter's primary store.
+    /// Returns true if operation is successful.
+    public entry fun burn(minter: &signer, amount: u64) acquires FiatTokenConfig, FiatTokenAccessInfo {
+        // SECURITY TODO: check if contract is paused 
+        // SECURITY TODO: check if transaction authorizer (signer) is a minter OR owner
+
+        // check if `minter` is blacklisted
+        assert!(!is_blacklisted(signer::address_of(minter)), error::permission_denied(EBLACKLISTED_ADDRESS));
+
+        // SECURITY TODO: check minter balance >= amount
+        assert!(balance_of_primary_store(signer::address_of(minter)) < amount, error::permission_denied(EINSUFFICIENT_BALANCE));
+        
+        // retrieve the refs using the `Metadata` object
+        let metadata_object = get_metadata_object();
+        let fiat_token_config = authorized_borrow_refs(minter, metadata_object);
+
+        // burn the tokens from the minter's primary store
+        let from_wallet = primary_fungible_store::primary_store(signer::address_of(minter), metadata_object);
+        fungible_asset::burn_from(&fiat_token_config.burn_ref, from_wallet, amount);
+
+        // TODO: emit burning event (??) or is FA events sufficient, indexing could be done that way. any other events?
+    }
+    
+    /// Transfer fiat tokens from `from` address to `to` address.
+    /// Authorizer can be any signer with sufficient balance in their primary store.
+    public entry fun transfer(from: &signer, to: address, amount: u64) acquires FiatTokenAccessInfo {
+        // SECURITY TODO: check if contract is paused
+
+        // check if `from` or  `to` address is blacklisted
+        assert!(!is_blacklisted(signer::address_of(from)), error::permission_denied(EBLACKLISTED_ADDRESS));
+        assert!(!is_blacklisted(to), error::permission_denied(EBLACKLISTED_ADDRESS));
+
+        // check if amount is greater than `from` address's balance
+        let from_addr = signer::address_of(from); 
+        assert!(balance_of_primary_store(from_addr) < amount, error::permission_denied(EINSUFFICIENT_BALANCE));
```

Comment:
Similarly `balance >= amount`.  Though with a lot of these checks, the fungible asset will return a similar error